src/compiler/transformers/destructuring.ts

```diff
@@ -63,9 +63,10 @@ namespace ts {
 
         function emitAssignment(name: Identifier, value: Expression, location: TextRange) {
             const expression = createAssignment(name, value, location);
-            if (isSimpleExpression(value)) {

```

Comment:
I plan to use it once we have verified all of the source map tests. The old emitter (and conversely, the new emitter) overzealously turns off source maps for entire subtrees when destructuring. The references to this function were removed to align with the old emitter, but will be restored once we have verified all of the source maps are correct and can isolate failing tests for individual changes.